### PR TITLE
[wasm] Fix cos and tan for large float numbers

### PR DESCRIPTION
**Pull request by chunnienc**
wasm fix for #7618

To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

### CODE REVIEW

File: tfjs-backend-wasm/src/cc/sin_cos_workaround.cc [hunk 1]
```diff
@@ -11,35 +11,67 @@
...
+
+  if (x < M_PI_4) {
+    return std::sin(x);
+  } else if (x < M_PI_2) {
+    return std::cos(M_PI_2 - x);
+  } else if (x < M_PI) {
+    return SinFixedImpl<T, /*is_shifted=*/true>(M_PI - x);
   } else {
-    return -sin_fixed(2 * M_PI - zero_to_2pi);
+    return -SinFixedImpl<T, /*is_shifted=*/true>(2 * M_PI - x);
```
Comment (mattsoulanille) [inline tfjs-backend-wasm/src/cc/sin_cos_workaround.cc]: [SinFixedImpl is missing](https://source.cloud.google.com/results/invocations/80cf68bf-f174-42b0-a8d6-d7c31381f6ad/targets/%2F%2Ftfjs-backend-wasm:tfjs-backend-wasm_pkg/log). Is there a missing file in this PR?

File: tfjs-backend-wasm/src/cc/sin_cos_workaround.cc [hunk 2]
```diff
@@ -11,35 +11,67 @@
...
   }
```
Comment (mattsoulanille) [inline tfjs-backend-wasm/src/cc/sin_cos_workaround.cc]: LGTM on the math.

File: tfjs-backend-wasm/src/cc/sin_cos_workaround.cc [hunk 3]
```diff
@@ -11,35 +11,67 @@
...
+
+  if (x < M_PI_4) {
+    return std::sin(x);
+  } else if (x < M_PI_2) {
+    return std::cos(M_PI_2 - x);
+  } else if (x < M_PI) {
+    return SinZeroTo2PI<T, /*is_shifted=*/true>(M_PI - x);
   } else {
-    return -sin_fixed(2 * M_PI - zero_to_2pi);
+    return -SinZeroTo2PI<T, /*is_shifted=*/true>(2 * M_PI - x);
```
Comment (mattsoulanille) [inline tfjs-backend-wasm/src/cc/sin_cos_workaround.cc]: What does the templated "is_shifted" value do? I'm not an expert at c++ templating.

**Review by mattsoulanille — APPROVED** [review]
LGTM. You can run `bazel test //tfjs-backend-wasm:bs_ios_12_tfjs-backend-wasm_test` to test on ios 12.

File: tfjs-backend-wasm/src/cc/sin_cos_workaround.cc (hunk 3 above)
Comment (chunnienc) [reply tfjs-backend-wasm/src/cc/sin_cos_workaround.cc]: Removed in the latest commit

**Pull request merged**